src/main/java/frc/robot/subsystems/Conveyor.java

```diff
@@ -139,6 +185,96 @@ public void stop() {
         back.stopMotor();
     }
 
+    @Override
+    public void periodic() {
+        previousColorSensorMeasurement = colorSensorMeasurement;
+        colorSensorMeasurement = colorSensor.getColorString();
+        lidarDistance = lidar.getAverageDistance();
+        colorSensorDistance = colorSensor.getProximity();
+
+        if (!doesBallExist()) {
+
+            frontState = ConveyorState.NONE;
+            backState = ConveyorState.NONE;
+
+        } else if (colorSensorMeasurement != previousColorSensorMeasurement) {
+
+            if (colorSensor.getColorString() == "Blue"
+                && MIN_COLOR_SENSOR_DISTANCE <= colorSensorDistance
+                && colorSensorDistance <= MAX_COLOR_SENSOR_DISTANCE) {
+
+                backState = frontState;
+                frontState = ConveyorState.BLUE;
+
+            } else if (colorSensor.getColorString() == "Red"
+                && MIN_COLOR_SENSOR_DISTANCE <= colorSensorDistance
+                && colorSensorDistance <= MAX_COLOR_SENSOR_DISTANCE) {
+
+                backState = frontState;
+                frontState = ConveyorState.RED;
+
+            } else if (!isBallQueued()) {
+
+                backState = frontState;
+                frontState = ConveyorState.NONE;
+
+            }
+
+        } else if (!isBallQueued()) {
+
+            backState = frontState;
+            frontState = ConveyorState.NONE;
+        }
+    }
+
+    /**
+     * Returns the type of ball present in the position closer to the intake.
+     *
+     * @return the type of ball present in the position closer to the intake
+     */
+    public ConveyorState getFront() {
```

Comment:
back and front are confusing, change to somthing like belt and index or somthing that does not rely on a consistant view of witch side of the robot is the front